simde/x86/avx2.h

```diff
@@ -2842,6 +2842,80 @@ simde_mm256_sllv_epi64 (simde__m256i a, simde__m256i b) {
   #define _mm256_sllv_epi64(a, b) simde_mm256_sllv_epi64(a, b)
 #endif
 
+SIMDE_FUNCTION_ATTRIBUTES
+simde__m256i
+simde_mm256_sra_epi16 (simde__m256i a, simde__m128i count) {
+  simde__m256i_private
+    r_,
+    a_ = simde__m256i_to_private(a);
+  simde__m128i_private
+    count_ = simde__m128i_to_private(count);
+
+  uint64_t shift = HEDLEY_STATIC_CAST(uint64_t, count_.i64[0]);
+
+  if (shift > 15) shift = 15;
+
+  #if defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
+    r_.i16 = a_.i16 >> HEDLEY_STATIC_CAST(int16_t, shift);
+  #else
+    SIMDE_VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.i16) / sizeof(r_.i16[0])) ; i++) {
+      r_.i16[i] = a_.i16[i] >> shift;
+    }
+  #endif
+
+  return simde__m256i_from_private(r_);
+}
+#if defined(SIMDE_X86_AVX2_NATIVE)
```

Comment:
These should go in the functions, instead of being implemented as macros.  The only time functions like this should be implemented as macros is when one of the arguments must be an immediate-mode constant (in which case the compiler will emit an error if we try to use it with a "normal" variable in a function, often even if that value is still always known at compile time).